.github/workflows/ci.yml

```diff
@@ -220,44 +220,42 @@ jobs:
     runs-on: ubuntu-22.04
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
 
     - name: Setup PHP
       uses: shivammathur/setup-php@v2
       with:
-        php-version: '7.4'
-        extensions: mbstring, intl, apcu, memcached, redis
-        tools: cs2pr
+        php-version: '8.1'
+        extensions: mbstring, intl
         coverage: none
-
-    - name: Get composer cache directory
-      id: composer-cache
-      run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-
-    - name: Get date part for cache key
-      id: key-date
-      run: echo "date=$(date +'%Y-%m')" >> $GITHUB_OUTPUT
-
-    - name: Cache composer dependencies
-      uses: actions/cache@v3
+        tools: phive
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+ 
+    - name: Phive install
+      uses: szepeviktor/phive@v1
       with:
-        path: ${{ steps.composer-cache.outputs.dir }}
-        key: ${{ runner.os }}-composer-${{ steps.key-date.outputs.date }}-${{ hashFiles('composer.json') }}-${{ matrix.prefer-lowest }}
+        home: "${{ runner.temp }}/.phive"
+        binPath: "${{ github.workspace }}/tools/phive"
 
-    - name: Composer install
-      run: composer stan-setup
+    - name: Install PHP tools with phive.
+      uses: szepeviktor/phive@v1
+      with:
+        home: "${{ runner.temp }}/.phive"
+        binPath: "${{ github.workspace }}/tools/phive"
+        trustGpgKeys: "CF1A108D0E7AE720, 12CE0F1D262429A5"
 
-    - name: Run PHP CodeSniffer
+    - name: Run phpcs
       run: vendor/bin/phpcs --report=checkstyle src/ tests/ | cs2pr
```

Comment:
still need phpcs installed.

Reply:
Oops.

Reply:
We could just cop the shared cs-stan workflow from 5.x into a 4.x branch and use it here.

https://github.com/cakephp/.github/blob/5.x/.github/workflows/cs-stan.yml

Reply:
But that workflow isn't using phive to install tools yet. There are additional copies of tool version numbers in those workflow steps. I'd like to not have to maintain versions in two places.